libs/sdk-ui-dashboard/src/model/commandHandlers/dashboard/common/insightReferences.ts

```diff
@@ -0,0 +1,22 @@
+// (C) 2021-2023 GoodData Corporation
+import { ObjRef, isInsightWidget, isDrillToInsight, IDashboardLayout } from "@gooddata/sdk-model";
+import { walkLayout } from "@gooddata/sdk-backend-spi";
+
+export function insightReferences(layout?: IDashboardLayout): ObjRef[] {
+    const insightRefsFromWidgets: ObjRef[] = [];
+    if (layout) {
+        walkLayout(layout, {
+            widgetCallback: (widget) => {
+                if (isInsightWidget(widget)) {
+                    insightRefsFromWidgets.push(widget.insight);
+                    widget.drills.map((drill) => {
```

Comment:
This will be reported by Sonar because you are not using the `map` results. What about
```ts
widget.drills.filter(isDrillToInsight).forEach((drill) => {
    insightRefsFromWidgets.push(drill.target);
});
```

Reply:
Copy paste bug, repaired